.github/workflows/ci.yml

```
@@ -20,8 +20,10 @@ jobs:
       - uses: actions/checkout@v2
       - uses: dtolnay/rust-toolchain@stable
       - uses: Swatinem/rust-cache@v1
-      - run: timeout 5m cargo test --workspace --all-features --all-targets
-
+      - run: |
+          timeout 5m cargo test -- --skip rpc::tests --skip sequencer::tests
+          timeout 10m cargo test sequencer::tests -- --test-threads=1
+          timeout 15m cargo test rpc::tests -- --test-threads=1
```

If we'd split these into three separate steps like

```suggestion
      - name: Unit tests
        run: timeout 5m cargo test -- --skip rpc::tests --skip sequencer::tests
      
      - name: Sequencer tests
        run: timeout 10m cargo test sequencer::tests -- --test-threads=1
      
      - name: RPC tests
        run: timeout 15m cargo test rpc::tests -- --test-threads=1
```

... Then the test output might become more useful? I didn't however find every one of these in the build output, yet.

---

Definitely would make everything much more readable :+1:

---

They've mentioned performance / slow response times in one of the TG groups. I wonder if we shouldn't move the RPC and sequencer tests to manual mode until they've resolved it? (different, manual trigger only job)

Waiting 20minutes per CI run is.. painful. Thoughts?

---

Let's skip seq and rpc for the time being, as they're really necessary whenever there's cairo release or an rpc spec update. Eventually the rpc tests will not refer to an underlying live api, while for the sequencer client we can mock a local server that provides real replies captured from a live seq api (and those would be updated only once per cairo release/seq api update).